zerver/webhooks/linear/view.py

```diff
@@ -0,0 +1,167 @@
+
+    has_assignee = "assignee" in payload["data"]
+    if has_assignee:
+        to_add.append(f"Assignee: {payload['data']['assignee']['name'].tame(check_string)}")
+
+    status = payload["data"]["state"]["name"].tame(check_string)
+    to_add.append(f"Status: {status}")
+
+    if to_add:
+        message += f"\n{', '.join(to_add)}."
+
+    return message
+
+
+def get_issue_remove_body(payload: WildValue, event: str) -> str:
+    return ISSUE_REMOVE_TEMPLATE.format(
+        type="Issue" if event == "issue" else "Sub-Issue",
+        number=payload["data"]["number"].tame(check_int),
+        title=payload["data"]["title"].tame(check_string),
+        team_name=payload["data"]["team"]["name"].tame(check_string),
+    )
+
+
+def get_comment_create_or_update_body(payload: WildValue, event: str, action: str) -> str:
+    message = COMMENT_CREATE_OR_UPDATE_TEMPLATE.format(
+        user=payload["data"]["user"]["name"].tame(check_string),
+        action=action,
+        url=payload["url"].tame(check_string),
+        issue_title=payload["data"]["issue"]["title"].tame(check_string),
+    )
+    message += CONTENT_MESSAGE_TEMPLATE.format(message=payload["data"]["body"].tame(check_string))
+    return message
+
+
+def get_comment_remove_body(payload: WildValue, event: str) -> str:
+    return COMMENT_REMOVE_TEMPLATE.format(user=payload["data"]["user"]["name"].tame(check_string))
+
+
+def get_issue_or_sub_issue_message(payload: WildValue, event: str) -> str:
+    action = payload["action"].tame(check_string)
+    if action == "remove":
+        return get_issue_remove_body(payload, event)
+
+    return get_issue_created_or_updated_message(
+        event, payload, action="created" if action == "create" else "updated"
+    )
+
+
+def get_comment_message(payload: WildValue, event: str) -> str:
+    action = payload["action"].tame(check_string)
+    if action == "remove":
+        return get_comment_remove_body(payload, event)
+
+    return get_comment_create_or_update_body(
+        payload, event, "commented" if action == "create" else "updated comment"
+    )
+
+
+EVENT_FUNCTION_MAPPER: Dict[str, Callable[[WildValue, str], str]] = {
+    "issue": get_issue_or_sub_issue_message,
+    "sub_issue": get_issue_or_sub_issue_message,
+    "comment": get_comment_message,
+}
+
+IGNORED_EVENTS = ["IssueLabel", "Project", "ProjectUpdate", "Cycle", "Reaction"]
+
+ALL_EVENT_TYPES = list(EVENT_FUNCTION_MAPPER.keys())
+
+
+@webhook_view("Linear", notify_bot_owner_on_invalid_json=True, all_event_types=ALL_EVENT_TYPES)
+@has_request_variables
+def api_linear_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: WildValue = REQ(argument_type="body", converter=to_wild_value),
+    user_specified_topic: Optional[str] = REQ("topic", default=None),
+) -> HttpResponse:
+    event_type = get_event_type(payload)
+    if event_type is None:
+        return json_success(request)
+
+    subject = get_subject(user_specified_topic, event_type, payload)
```

Comment:
While we do have remnants of the use of `subject` here and there, we are slowly moving towards replacing them with `topic` (#1192). This applies to both the variable and the function name here.